modules/internal/ChapelIO.chpl

```diff
@@ -198,25 +493,54 @@ module ChapelIO {
       }
     }
   }
-  
+
+
+  /*
+     An abstract base class type for classes that know how
+     to read values.
+   */
   class Reader {
+    /* returns `false` */
     proc writing param return false;
-    // if it's binary, we don't decorate class/record fields and values
+    /* returns `true` if this Reader is configured for binary I/O. */
     proc binary():bool { return false; }
-    // get other style elements
+    /* return other style elements.
+       subclasses should override this method */
+    pragma "no doc"
     proc styleElement(element:int):int { return 0; }
 
+    /* 
+       Return any saved error code.
+       Subclasses should override this method.
+     */
     proc error():syserr { return ENOERR; }
```

Comment:
I'd like to merge these with channel, which naturally has a common "parent" because they're just different specializations of the same type. But I don't want to do that now.